core/src/main/java/org/infinispan/partitionhandling/impl/PreferConsistencyStrategy.java

```diff
@@ -179,10 +179,8 @@ public void onPartitionMerge(AvailabilityStrategyContext context,
          actualMembers.retainAll(mergedTopology.getMembers());
          mergedAvailabilityMode = AvailabilityMode.DEGRADED_MODE;
       } else {
-         log.debugf("No current topology, recovered only joiners for cache %s", context.getCacheName());
-         mergedTopology = null;
-         actualMembers = null;
-         mergedAvailabilityMode = AvailabilityMode.AVAILABLE;
+         log.debugf("No current topology, recovered only joiners for cache %s. Skipping availability update.", context.getCacheName());
+         return;
```

Comment:
I'm not sure about this change, I'm thinking maybe the coordinator already has a non-null topology in the `ClusterCacheStatus` and in that case, it should be replaced with a `null` topology.

Also, the test passes even without this change.